### PR TITLE
remove ctypedef class from lib.pyx

### PR DESCRIPTION
**Pull request by WillAyd**
@jbrockmendel

### CODE REVIEW

**Comment by jbrockmendel** [comment]
`E   TypeError: issubclass() arg 1 must be a class` matches what i got locally from the same approach

**Comment by WillAyd** [comment]
Ah ok. I guess this makes it so dtype is not a Python object, so probably need to change that check around

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -1961,7 +1945,7 @@ cdef class BytesValidator(Validator):
         return isinstance(value, bytes)
 
     cdef bint is_array_typed(self) except -1:
-        return issubclass(self.dtype.type, np.bytes_)
+        return self.dtype.type == cnp.NPY_BYTE or self.dtype.type == cnp.NPY_UBYTE
```
Comment (mroeschke) [inline pandas/_libs/lib.pyx:1948]: cc @seberg is this the right way to check for bytes?

Comment (seberg) [reply pandas/_libs/lib.pyx:1948]: No, its `NPY_STRING` (string is `NPY_UNICODE`, leftover from python 2)

File: pandas/_libs/lib.pyx [hunk 2]
```diff
@@ -1944,7 +1928,7 @@ cdef class StringValidator(Validator):
         return isinstance(value, str)
 
     cdef bint is_array_typed(self) except -1:
-        return issubclass(self.dtype.type, np.str_)
+        return cnp.PyDataType_ISSTRING(self.dtype)
```
Comment (seberg) [inline pandas/_libs/lib.pyx:1931]: So this is also wrong, `ISSTRING` checks for bytes (NPY_STRING) OR unicode.  You want `== NPY_UNICODE` here if you use this.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @WillAyd

**Comment by jbrockmendel** [comment]
nice!

**Pull request merged**